Platform/Ampere/buildfw.sh

```diff
@@ -0,0 +1,286 @@
+else
+  EDK2_HEAP_GUARD_ENABLE=TRUE
+fi
+
+UPD720202_ROM_FILE="K2026090.mem"
+EXTRA_BUILD_FLAGS=""
+if [ "${BOARD_NAME}" = "ComHpcAlt" ] && [ -e "${WORKSPACE}/${UPD720202_ROM_FILE}" ]; then
+  EXTRA_BUILD_FLAGS+=" -D USB_UPD720202_ROM_FILE=${WORKSPACE}/${UPD720202_ROM_FILE}"
+fi
+
+if [ -e "${WORKSPACE}/IntelUndiBin/Release/AARCH64/GigUndiDxe.efi" ]; then
+  EXTRA_BUILD_FLAGS+=" -D INTEL_UNDI_BIN=TRUE"
+fi
+
+build -a AARCH64 -t ${TOOLCHAIN} -b ${BLDTYPE} -n ${BUILD_THREADS} \
+        -D FIRMWARE_VER="${VER}-${BUILD} TF-A ${TFA_VERSION}"      \
+        -D MAJOR_VER=${MAJOR_VER} -D MINOR_VER=${MINOR_VER}        \
+        -D SECURE_BOOT_ENABLE=${EDK2_SECURE_BOOT_ENABLE}      \
+        -D NETWORK_ENABLE=${EDK2_NETWORK_ENABLE}              \
+        -D INCLUDE_TFTP_COMMAND=${EDK2_INCLUDE_TFTP_COMMAND}  \
+        -D NETWORK_IP6_ENABLE=${EDK2_NETWORK_IP6_ENABLE}      \
+        -D NETWORK_ALLOW_HTTP_CONNECTIONS=${EDK2_NETWORK_ALLOW_HTTP_CONNECTIONS} \
+        -D NETWORK_TLS_ENABLE=${EDK2_NETWORK_TLS_ENABLE}      \
+        -D REDFISH_ENABLE=${EDK2_REDFISH_ENABLE}              \
+        -D PERFORMANCE_MEASUREMENT_ENABLE=${EDK2_PERFORMANCE_MEASUREMENT_ENABLE} \
+        -D TPM2_ENABLE=${EDK2_TPM2_ENABLE}                    \
+        -D HEAP_GUARD_ENABLE=${EDK2_HEAP_GUARD_ENABLE}        \
+        -Y COMPILE_INFO -y BuildReport.log                         \
+        -p Platform/${MANUFACTURER}/${BOARD_NAME}Pkg/${BOARD_NAME}${LINUXBOOT}.dsc \
+        ${EXTRA_BUILD_FLAGS}
+
+OUTPUT_BASENAME=${OUTPUT_BIN_DIR}/${BOARD_NAME,,}_tfa_uefi_${BLDTYPE,,}_${VER}-${BUILD}
+
+OUTPUT_UEFI_IMAGE=Build/${BOARD_NAME}/${BOARD_NAME,,}_uefi_${BLDTYPE,,}_${VER}-${BUILD}.bin
+OUTPUT_TFA_UEFI_IMAGE=Build/${BOARD_NAME}/${BOARD_NAME,,}_tfa_uefi_${BLDTYPE,,}_${VER}-${BUILD}.bin
+OUTPUT_SPINOR_IMAGE=Build/${BOARD_NAME}/${BOARD_NAME,,}_rom_${BLDTYPE,,}_${VER}-${BUILD}.bin
+
+cp -v "Build/${BOARD_NAME}/${BLDTYPE}_${TOOLCHAIN}/FV/BL33_${BOARD_NAME^^}_UEFI.fd" "${OUTPUT_UEFI_IMAGE}"
+
+if [ -f "${TFA_SLIM}" ]; then
+  # Create a 2MB file with 0xff
+  dd bs=1024 count=2048 if=/dev/zero | tr "\000" "\377" > "${OUTPUT_TFA_UEFI_IMAGE}"
+  dd bs=1024 conv=notrunc if="${TFA_SLIM}" of="${OUTPUT_TFA_UEFI_IMAGE}"
+  dd bs=1 seek=2031616 conv=notrunc if="${OUTPUT_BOARD_SETTINGS_BIN}" of="${OUTPUT_TFA_UEFI_IMAGE}"
+fi
+
+rm -f "${OUTPUT_BOARD_SETTINGS_BIN}"
+rm -f "${OUTPUT_BIN_DIR}/$(basename ${BOARD_SETTINGS_CFG}).txt"
+
+if [ -f "${TFA_SLIM}" ]; then
+  # Write the UEFI image
+  dd bs=1024 seek=2048 if="${OUTPUT_UEFI_IMAGE}" of="${OUTPUT_TFA_UEFI_IMAGE}"
+
+  # Create the full SPI-NOR image
+  dd bs=1M count=${SPI_SIZE_MB} if=/dev/zero | tr "\000" "\377" > "${OUTPUT_SPINOR_IMAGE}"
+  # Write the code (TF-A+UEFI) area
+  dd bs=1M seek=4 conv=notrunc if="${OUTPUT_TFA_UEFI_IMAGE}" of="${OUTPUT_SPINOR_IMAGE}"
+
+  cp -vf "${OUTPUT_TFA_UEFI_IMAGE}" "Build/${BOARD_NAME}/${BOARD_NAME,,}_tfa_uefi.bin"
+
+  cp -vf "${SCP_SLIM}" "Build/${BOARD_NAME}/altra_scp.slim"
```

Comment:
If SCP_SLIM is not available, we should ignore it and complete building the ATF+EDK2 binary and flashing.

Reply:
Good catch!